qstarz_bl_1000.cc

```diff
@@ -0,0 +1,297 @@
+/*
+    Handle Qstarz BL-1000 .BIN files.
+
+    Saved in the SESSION/GPSLog folder. There will 3 files sharing the same base name: .BIN, .POI, and .DAT.
+    Only the .BIN file is of interest to us.
+
+    Copyright (C) 2020 Pierre Bernard, pierre.bernard@houdah.com
+    Copyright (C) 2001-2020 Robert Lipe, robertlipe+source@gpsbabel.org
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+ */
+
+#include <QtCore/QDataStream>
+#include <QtCore/QFile>
+
+#include "defs.h"
+#include "gbfile.h"
+#include "src/core/logging.h"   // for Warning, Fatal
+
+#include "qstarz_bl_1000.h"
+
+#define MYNAME "Qstarz BL-1000"
+
+
+enum BL1000_POINT_TYPE {
+  BL1000_POINT_TYPE_UNKNOWN = '-',
+  BL1000_POINT_TYPE_WAY_POINT = 'B', // Button push
+  BL1000_POINT_TYPE_TRACK_POINT_BY_TIME = 'T',
+  BL1000_POINT_TYPE_TRACK_POINT_BY_DISTANCE = 'D',
+  BL1000_POINT_TYPE_TRACK_POINT_BY_SPEED = 'S',
+};
+
+
+/***************************************************************************
+ *           local helper functions                                        *
+ ***************************************************************************/
+
+inline bool qstarz_bl_1000_is_waypoint_type(BL1000_POINT_TYPE type)
+{
+  return (type == BL1000_POINT_TYPE_WAY_POINT);
+}
+
+inline bool qstarz_bl_1000_is_trackpoint_type(BL1000_POINT_TYPE type)
+{
+  return (
+           (type == BL1000_POINT_TYPE_TRACK_POINT_BY_TIME) ||
+           (type == BL1000_POINT_TYPE_TRACK_POINT_BY_DISTANCE) ||
+           (type == BL1000_POINT_TYPE_TRACK_POINT_BY_SPEED)
+         );
+}
+
+void
+QstarzBL1000Format::qstarz_bl_1000_read(QDataStream& stream)
+{
+  route_head* track_route = new route_head;
```

Comment:
We have been running some clang-tidy checks lately, and implementing a few.  In the case were the type is obvious auto makes sense.  I would suggest
auto* track_route = new route_head;

> qstarz_bl_1000.cc:68:3: warning: use auto when initializing with new to avoid duplicating the type name [modernize-use-auto]
>   route_head* track_route = new route_head;
>   ^~~~~~~~~~~
>   auto